$NetBSD: patch-aa,v 1.3 2005/03/11 21:27:17 rillig Exp $

This patch disables the priority scheduling if it is not available.

Needed at least for NetBSD-1.6.2.

--- src/DarkIce.cpp.orig	Sun Feb 15 13:14:38 2004
+++ src/DarkIce.cpp	Fri Mar 11 18:01:41 2005
@@ -873,6 +873,7 @@ DarkIce :: configFileCast (  const Confi
 void
 DarkIce :: setRealTimeScheduling ( void )               throw ( Exception )
 {
+#ifdef _POSIX_PRIORITY_SCHEDULING
     uid_t   euid;
 
     euid = geteuid();
@@ -917,6 +918,7 @@ DarkIce :: setRealTimeScheduling ( void 
         reportEvent( 1,
         "It is recommended that you run this program as super-user");
     }
+#endif
 }
 
 
@@ -928,6 +930,7 @@ DarkIce :: setRealTimeScheduling ( void 
 void
 DarkIce :: setOriginalScheduling ( void )               throw ( Exception )
 {
+#ifdef _POSIX_PRIORITY_SCHEDULING
     uid_t   euid;
 
     euid = geteuid();
@@ -947,6 +950,7 @@ DarkIce :: setOriginalScheduling ( void 
 
         reportEvent( 5, "reverted to original scheduling");
     }
+#endif
 }
 
 
